src/internet_identity_interface/src/internet_identity/types.rs

```diff
@@ -24,6 +25,13 @@ pub struct DeviceData {
     pub key_type: KeyType,
     pub protection: DeviceProtection,
     pub origin: Option<String>,
+    // Metadata map for additional device information.
+    //
+    // Note: some fields above will be moved to the metadata map in the future.
+    // If the metadata map already contains a field with the same name, the explicit
+    // field will be dropped and the data will be lost!
```

Comment:
Not sure I understand this comment: what is meant by "a field with the same name", and that the explicit field will be dropped? If I insert to metadata key-value pair ("alias", some_entry), then the value of the field `DeviceData.alias` will be "dropped"?   What does "dropped" mean in this case?

Reply:
Actually, let's not just have a comment but enforce this using a canister side check. This way we'll be sure that no data is lost in an upcoming migration.

Reply:
Now the intention is clear, and I do agree that the enforcing is the right way to go. Thanks!